conda_smithy/templates/azure-pipelines-win.yml.tmpl

```diff
@@ -42,10 +42,10 @@ jobs:
 
     - task: CondaEnvironment@1
       inputs:
-        packageSpecs: 'python=3.6 conda-build conda conda-forge::conda-forge-ci-setup=3 pip' # Optional
```

Comment:
you can't remove conda-build here. How are we going to build packages?

Reply:
also please leave python at 3.6

Reply:
@beckermr conda-build is a dependency of conda-forge-ci-setup

Reply:
Having conda-forge-ci-setup handle the dependencies (as it's name would entail) in a single source is better than maintaining a depedency list across templates, imo

Reply:
But it could not be in the future. We cannot rely on deps of deps to install what we need.

Reply:
That seems to go against shifting the ci setup & functionality to the setup package in the first place? Should all of it's dependencies also be listed here explicitly too? I agree with the last PR that depending on the dependencies of conda-build isn't right, but here I'm not sure I agree.

Reply:
The CI templates explicitly call conda build, so they need to explicitly list it as a dep. Ditto for pip in the case where a local package is used.

Reply:
I don't disagree with that. The ci-setup-package runtime depends on conda-build & pip, yet the package itself doesn't call conda build or pip. If it is not intended to be the source of the runtime dependencies of the templates then I would suggest they are removed?

Reply:
We use conda build and pip in the templates so we should install them. The fact that the ci setup package depends on them too is somewhat incidental. 

IDK what the intent of that package was originally and I don't think it matters. 

By making sure to install the things we use in the template itself, we decouple smithy from the ci setup package more. This is better in the long term since we are then free to make changes in one without breaking the other.